### PR TITLE
fix: replace force-unwraps with guards in TextDecoder.decodeText

### LINKED ISSUE #414: TextDecoder.decodeText force-unwraps cause crashes during repeated live transcription
## Summary

`TextDecoder.decodeText` contains force-unwraps that crash the app during repeated rapid transcription calls (e.g., live transcription preview that re-transcribes every ~200ms).

## Crash Stack

```
EXC_BREAKPOINT / SIGTRAP
_assertionFailure → TextDecoder.decodeText(from:using:sampler:options:callback:)
→ TranscribeTask.run → WhisperKit.transcribe → WhisperKit.transcribe(audioPath:)
```

## Root Cause

Two force-unwraps in `TextDecoder.decodeText` ([TextDecoder.swift](https://github.com/argmaxinc/WhisperKit/blob/main/Sources/WhisperKit/Core/TextDecoder.swift)):

1. **Line 747**: `decoderInputs.initialPrompt.last!` — crashes if `initialPrompt` is empty
2. **Line 829**: `decoderOutput.logits!` — crashes if logits is nil

These are fine for single-shot transcription but become problematic during sustained rapid calls where internal state can degrade.

## Reproduction

- Create two WhisperKit instances (main + live)
- Call `transcribe(audioPath:)` on the live instance in a tight loop (~5-10 calls/sec) during recording
- After extended use (minutes to hours), the decoder state accumulates corruption and one of the force-unwraps hits nil

## Environment

- WhisperKit v0.15.0
- macOS 26.2, Apple Silicon
- Model: openai_whisper-small
- Using separate WhisperKit instances for main and live transcription

## Suggested Fix

Replace the force-unwraps with guard statements that throw errors instead of crashing:

```swift
// Line 747
guard let firstToken = decoderInputs.initialPrompt.last else {
    throw WhisperError.decodingFailed("initialPrompt is empty")
}
var nextToken: Int = firstToken

// Line 829
guard let logits = decoderOutput.logits else {
    throw WhisperError.decodingLogitsFailed("Logits output is nil")
}
```

This would allow callers to handle the error gracefully instead of crashing the entire app.

## Related

I noticed #413 and #412 are addressing thread safety — this may be related to the same class of issues.

### PR DESCRIPTION
**Pull request by yw5aj**
Fixes #414

Two force-unwraps in `decodeText` crash the app when decoder state degrades during sustained rapid transcription calls (e.g., live transcription preview at ~5 calls/sec):

- **Line 747**: `decoderInputs.initialPrompt.last!` — crashes when `initialPrompt` is empty
- **Line 829**: `decoderOutput.logits!` — crashes when logits is nil

This PR replaces them with guard statements that throw descriptive errors, allowing callers to handle the failure gracefully instead of crashing the entire app.

### Minimal diff — no behavioral change for normal operation
The guard statements only trigger when the values are unexpectedly nil. Normal transcription flow is unaffected.

### CODE REVIEW

**Comment by ZachNagengast** [comment]
Interesting, third PR this week for this specific issue #420 #417. I see you are the author of the original issue, do you have any relation to the other two authors?

**Comment by yw5aj** [comment]
No relation — we're an independent team that hit the same crash during live transcription. Glad to see others confirming the issue. Happy to defer to whichever PR you prefer to merge\!

**Comment by ZachNagengast** [comment]
👍󠄳󠅜󠅟󠅣󠅕

**Comment by achyutbenz19** [comment]
Took an independent look at the diff. A couple of things that might be useful.

**1. Rebase is clean.** The branch is on `2870b46`, and main has moved 6 commits ahead, most notably #456 (the package rename) and #455 (the swift-transformers to in-tree Hub/Tokenizers switch). `git rebase main` applies cleanly with no conflicts. Post-rebase, the patch is a pure +8/-2 on `Sources/WhisperKit/Core/TextDecoder.swift`. Confirmed with Swift 6.2 / Xcode 26 on Apple Silicon: the rebased HEAD builds, and the model-free TextDecoder tests (`testTextDecoderMLMultiArrayInputType`, `testTextDecoderMLMultiArrayOutputType`, `testTextDecoderMLMultiArrayOutputTypeWithNilValues`, `testDecodingCacheInitialization`, `testDecodingCacheWithNilValues`, `testMLMultiArrayConformsToTextDecoderTensorType`) pass 6/6 on both `main` and the rebased commit. Commands:

```
git fetch origin
git checkout fix/safe-unwrap-text-decoder
git rebase origin/main
```

**2. Adjacent pattern, noting for your judgment.** Lines 843 and 844 of current `main` (`sampleResult.tokens.last!`, `sampleResult.logProbs.last!`) are the same shape as the two unwraps this PR fixes. #420 covers them, this PR does not. Not suggesting you expand scope, just flagging in case it changes the merge calculus.

Disclosure: I am an AI assistant (Anthropic's Claude) helping a user contribute to WhisperKit. I did not reproduce the crash deterministically, it depends on rare runtime states that a file fixture will not reliably hit, so this is build and test signal only, not before / after empirical evidence.
